
# Week – 1: Simple linear regression

## Problem Statement
To build a regression model that predicts the housing prices of a new house using the Boston Housing dataset.


## Dataset Description

The given **Boston Housing dataset** is provided as a CSV file containing **506 observations(rows)** and **14 attributes(columns)** describing various socio-economic, environmental, and housing characteristics of different areas in Boston.

Each observation includes variables such as crime rate, number of rooms, pollution levels (NOX), and the median value of owner-occupied homes (MEDV).

For this experiment, we focus on only **two variables**:

- **NOX** – Nitric oxide concentration (parts per 10 million)
- **MEDV** – Median value of owner-occupied homes (in $1000s)

The objective of this experiment is to **predict the median value of homes (MEDV)** using nitric oxide concentration (NOX) as the predictor variable or dependent variable.

## Technique: Simple Linear Regression

<img src="img01.png" alt="Linear Regression Line" width="600"><br>

Simple linear regression is a method used to study the relationship between one input variable and one output variable by fitting a straight line to the data.

In this experiment, **NOX** (nitric oxide concentration) is the input variable, and **MEDV** (median house price) is the output variable. The goal is to predict house prices based on pollution levels.

The model follows the equation:

**MEDV = β₀ + β₁(NOX) + ε**

Here:
- **β₀** is the intercept (the predicted value when NOX is 0),
- **β₁** is the slope (how much MEDV changes when NOX increases by one unit),
- **ε** represents the error (the difference between actual and predicted values).

The most important part of this model is the slope, because it tells us whether pollution increases or decreases house prices, and by how much.

This method is easy to understand and interpret, but it assumes that the relationship between NOX and MEDV is linear. It also assumes that errors are evenly spread and independent. If these assumptions are not satisfied, the model’s predictions may not be fully reliable.

## Procedure | Experiment 

In [1]:
import pandas as pd 
from sklearn.linear_model import LinearRegression
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error,r2_score,root_mean_squared_error

df = pd.read_csv("Boston.csv")
df=df.iloc[:,1:]

X = df[["nox"]]
y = df["medv"]


X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2,random_state=42)


model=LinearRegression()
model.fit(X_train,y_train)
y_predict = model.predict(X_test)


print(r2_score(y_test,y_predict))
print(mean_absolute_error(y_test,y_predict))
print(root_mean_squared_error(y_test,y_predict))

0.2052049761980269
5.695364590575609
7.634482689246051


## Results 
NOX appears to be a weak standalone predictor of housing value, indicating that house prices are determined by a combination of factors rather than pollution levels alone.

- The model achieved an R² value of 0.205, indicating that approximately 20.5% of the variance in median housing prices is explained by nox.
- The mean absolute error of 5.69 suggests that, on average, predicted prices differ from actual prices by about $5,690(1 unit in column medv equals 1000 dollars as per dataset). 
<br>
